tools/ci_build/github/mac-ios-swift-packaging-pipeline.yml

```diff
@@ -0,0 +1,117 @@
+parameters:
+- name: BuildType
+  displayName: |-
+    Type of build.
+    "normal": A normal build not for publication.
+  type: string      #TODO: add other build types here
+  values:
+  - normal
+  default: normal
+
+name: "$(Date:yyyyMMdd)$(Rev:rrr)"  # build number format
+
+stages:
+- stage: SPM_IOS_Local_Packaging_Testing
+  dependsOn: []
+  jobs:
+  - job: SPMIosPackaging
+    displayName: "SPM iOS Packaging Local"
+
+    pool:
+      vmImage: "macOS-13"
+
+    variables:
+      xcodeVersion: "14.3"
+      artifactsName: "ios_packaging_artifacts" #TODO: Add `_full` suffix when syncing to latest main
+
+    timeoutInMinutes: 300
+
+    steps:
+    - template: templates/use-xcode-version.yml
+      parameters:
+        xcodeVersion: ${{ variables.xcodeVersion }}
+
+    # Download artifacts from a specific pipeline 
+    # For now, it consumes rel-1.15.0 version ORT iOS Pod which matches the current source code
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: 'Lotus'
+        definition: 995
+        buildVersionToDownload: 'latest'
+        branchName: 'rel-1.15.0'
```

Comment:
Should this be latest from `main`? I would have expected we'd use the pod from the Azure blob storage for a release but maybe this was just for initial testing.

Does this pull from the latest successful build, or will this CI fail if the latest build of the ORT pipeline failed and the artifact wasn't published?

Reply:
> Should this be latest from main? I would have expected we'd use the pod from the Azure blob storage for a release but maybe this was just for initial testing.

Currently the objc source code to be checked in is in sync with the `rel-1.15.0` branch, so also here uses the pod archive from the rel-1.15.0 branch. After we pull the latest main changes here, will update the branchName to be `main`. 
Added a minor comment here as syncing to latest main will also need some changes about how we find pod artifacts in ios-packaging pipelines, etc. https://github.com/microsoft/onnxruntime-swift-package-manager/blob/6dcdf0b6b47c5bf1cbc871820248aa58e0a461e6/tools/ci_build/github/mac-ios-swift-packaging-pipeline.yml#L25
will add to-do comments here too.


> Does this pull from the latest successful build, or will this CI fail if the latest build of the ORT pipeline failed and the artifact wasn't published?

It will pull from the latest successful run from the specified pipeline runs.

Reply:
> Does this pull from the latest successful build, or will this CI fail if the latest build of the ORT pipeline failed and the artifact wasn't published?

It will pull from the latest successful run from the specified pipeline runs.